src/types/did_url.rs

```diff
@@ -117,20 +144,39 @@ impl DidUrl {
     pub fn parse<S: AsRef<str>>(input: S) -> Result<Self, Error> {
         let url = Url::parse(input.as_ref())?;
 
-        let mut path = url.path().split('/');
+        // Note that `url.path()` will return an incorrect path from did url
```

Comment:
we could move parsing of query and fragment into `parse_ethr_did` to stay concise here, which would also completely remove our dependency on `url`